### ISSUE TITLE
[table]: Allow exanding rows, redrawing only the toggled row

### ISSUE THREAD

**Issue by ctron** [patternfly-yew/patternfly-yew#69]
Currently we do have a full state of the table. Toggling a table entry forces the full table to be re-drawn. This should be more efficient and only toggled rows should be re-rendered.

**Comment by phill-85** [comment]
Have you tried using keys? https://yew.rs/docs/concepts/html/lists#keyed-lists

**Comment by ctron** [comment]
Yes, the current implementation is actually using them. However, that still means that the whole table needs to be re-processes.

I would hope that there's a solution where each row becomes a component, and manages its own state. The downside of that might be that there is no (table) global state for expansions. Which might be ok in most scenarios I guess.

Then again, I guess we need to split up the current table component into smaller bits, and follow the idea of "composition" to build different ways of dealing with tables. Maybe having different approaches for different use cases.

**Comment by aDogCalledSpot** [comment]
#128 addresses the composition issue. So implementing this should be possible now. However, `Table` doesn't currently use state internally per row so that would still need to be implemented.
